### ISSUE TITLE
redshift dialect: parse error while casting `AS information_schema.sql_identifier`

### ISSUE THREAD

**Issue by 6e2f61** [sqlfluff/sqlfluff#3519]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Querying redshift system tables produces parsing error while casting.

### Expected Behaviour

Add support for the redshift system table select query syntax.

### Observed Behaviour

Part of query not currently supported, results in parse errors while casting with dot character.

```bash
== [app/sys_tbl_redshift_part2.sql] FAIL                                                                                                            
L:   1 | P:   1 | L036 | Select targets should be on a new line unless there is only one select target.
L:   2 | P:  30 |  PRS | Line 2, Position 30: Found unparsable section: ' AS information_schema.sql_identifier'
WARNING: Parsing errors found and dialect is set to 'redshift'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Create a test file containing as below and try to lint/parse with sqlfluff.

```sql
SELECT
    CAST(pg_namespace.nspname AS information_schema.sql_identifier) AS constraint_schema
FROM
    pg_namespace
```

### Dialect

redshift

### Version

`sqlfluff, version 1.0.0`
`Python 3.8.10`

### Configuration

```bash
# https://docs.sqlfluff.com/en/stable/configuration.html#config

[sqlfluff]
dialect = redshift
templater = jinja
rules = all
exclude_rules = L034
# Controls SQLFluff output, see max_line_length for SQL output
output_line_length = 118
ignore_templated_areas = true
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = false
indented_using_on = true
template_blocks_indent = true

[sqlfluff:rules]
tab_space_size = 4
max_line_length = 118
indent_unit = space
comma_style = trailing

# https://docs.sqlfluff.com/en/stable/rules.html#ruleref

[sqlfluff:rules:L003]
hanging_indents = true

[sqlfluff:rules:L007]
operator_new_lines = after

[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = upper

[sqlfluff:rules:L011]
# of tables
aliasing = explicit

[sqlfluff:rules:L012]
# of columns
aliasing = explicit

[sqlfluff:rules:L013]
allow_scalar = false

[sqlfluff:rules:L014]
# Unquoted identifiers
extended_capitalisation_policy = lower

[sqlfluff:rules:L016]
ignore_comment_clauses = true
ignore_comment_lines = true
indent_unit = space
max_line_length = 160
tab_space_size = 4

[sqlfluff:rules:L019]
comma_style = leading

[sqlfluff:rules:L022]
comma_style = leading

[sqlfluff:rules:L029]
unquoted_identifiers_policy = all
ignore_words = character_maximum_length, column_name, columns, constraint_name, constraint_schema, data_type, role, status, table_name, type

[sqlfluff:rules:L030]
extended_capitalisation_policy = upper

[sqlfluff:rules:L031]
# Avoid table aliases in from clauses and join conditions
force_enable = true

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]
capitalisation_policy = upper

[sqlfluff:rules:L051]
fully_qualify_join_types = both

[sqlfluff:rules:L052]
multiline_newline = false
require_final_semicolon = false

[sqlfluff:rules:L054]
group_by_and_order_by_style = explicit

[sqlfluff:rules:L057]
additional_allowed_characters = "{}"
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
allow_space_in_identifier = false

[sqlfluff:rules:L059]
prefer_quoted_identifiers = false

[sqlfluff:rules:L063]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Isn’t this a duplicate of #3512 ?

**Comment by 6e2f61** [comment]
Actually I didn't know if I need to report 2 separate errors, or put it in one issue.
For the same query (full one) there are 2 different parsing errors, so I decided to raise 2 separate issues. If it is not necessary Let's close this one and I will update the other one.

**Comment by tunetheweb** [comment]
Happy to keep this open if you can rename the other one. And ideally reduce the huge test case to a minimally viable one that just produces that other error.

**Comment by 6e2f61** [comment]
@tunetheweb done. Thanks for the tip.
